src/main/scala/cromwell/engine/backend/jes/JesAttributes.scala

```diff
@@ -0,0 +1,41 @@
+package cromwell.engine.backend.jes
+
+import java.net.URL
+
+import com.typesafe.config.{Config, ConfigFactory}
+import cromwell.util.ConfigUtil._
+
+case class JesAttributes(applicationName: String,
+                         project: String,
+                         executionBucket: String,
+                         endpointUrl: URL,
+                         authMode: GcsAuthMode,
+                         docker: Option[DockerAuthInformation])
+object JesAttributes {
+
+  def apply(): JesAttributes = {
+    val jesConf = ConfigFactory.load.getConfig("backend").getConfig("jes")
+    val refConf: Config = ConfigFactory.parseResources("jes.conf")
+
+    jesConf.checkValidWrapped(refConf, "Jes")
+    jesConf.warnNotRecognized(refConf, "Jes", "dockerAccount", "dockerToken")
+
+    val applicationName = jesConf.getString("applicationName")
+    val project = jesConf.getString("project")
+    val executionBucket = jesConf.getString("baseExecutionBucket")
+    val endpointUrl = jesConf.getURL("endpointUrl")
+    val authMode = GcsAuthMode.fromString(jesConf.getString("authenticationMode"))
+    val docker = for {
+      account <- jesConf.getStringOption("dockerAccount")
+      token <- jesConf.getStringOption("dockerToken")
+    } yield DockerAuthInformation(account, token)
+
```

Comment:
Need moar types ! :) Could be an option yes, although I think it's doesn't address the "validation" part in terms of "is this a boolean ?", "is this a correct url ?" etc...
I also agree that it's not clear as the special cases stuff is confusing. Let me think about it see if I can find a way to clarify it.